cirq/value/value_equality.py

```diff
@@ -73,33 +91,62 @@ def _value_equality_hash(self: _SupportsValueEquality) -> int:
                  self._value_equality_values_()))
 
 
+def _value_equality_approx_eq(self: _SupportsValueEquality,
+                              other: _SupportsValueEquality,
+                              atol: float) -> bool:
+    # Preserve regular equality type-comparison logic.
+    cls_self = self._value_equality_values_cls_()
+    if not isinstance(other, cls_self):
+        return NotImplemented
+    cls_other = other._value_equality_values_cls_()
+    if cls_self != cls_other:
+        return False
+
+    # Delegate to cirq.approx_eq for approximate equality comparison.
+    return cirq.approx_eq(
+        self._value_equality_approximate_values_(),
+        other._value_equality_approximate_values_(),
+        atol=atol
+    )
+
+
+
 # pylint: disable=function-redefined
 @overload
 def value_equality(cls: type,
                    *,
                    unhashable: bool = False,
-                   distinct_child_types: bool = False
+                   distinct_child_types: bool = False,
+                   approximate: bool = False,
                    ) -> type:
     pass
 
 
 @overload
 def value_equality(*,
                    unhashable: bool = False,
-                   distinct_child_types: bool = False
+                   distinct_child_types: bool = False,
+                   approximate: bool = False,
                    ) -> Callable[[type], type]:
     pass
 
 
 def value_equality(cls: type = None,
                    *,
                    unhashable: bool = False,
-                   distinct_child_types: bool = False
+                   distinct_child_types: bool = False,
+                   approximate: bool = False,
                    ) -> Union[Callable[[type], type], type]:
     """Implements __eq__/__ne__/__hash__ via a _value_equality_values_ method.
 
     _value_equality_values_ is a method that the decorated class must implement.
 
+    _value_equality_approximate_values_ is a method that the decorated class
+    might implement if special support for approximate equality is required.
+    This is only used when approximate argument is set. When approximate
+    argument is set and _value_equality_approximate_values_ is not defined,
```

Comment:
Give an example of how it would be used, in terms of angles.

Reply:
Done; Invented new name for angles comparator, WDYT?

Reply:
I'm thinking about changing this name to `PeriodicEquivalence` actually.

Reply:
PeriodicValue is better.